R/generateSettings.R

```diff
@@ -56,34 +65,55 @@ generateSettings <- function(standard="None", chart="eDish"){
   )
     
+  potential_settings <- settings
+  
   if(tolower(standard)=="adam"){
-    settings[["id_col"]]<-"USUBJID"
-    settings[["value_col"]]<-"AVAL"
-    settings[["measure_col"]]<-"PARAM"
-    settings[["normal_col_low"]]<-"A1LO"
-    settings[["normal_col_high"]]<-"A1HI"
-    settings[["studyday_col"]]<-"ADY"
-    settings[["visit_col"]]<-"VISIT"
-    settings[["visitn_col"]]<-"VISITNUM"
-    settings[["measure_values"]][["ALT"]]<-"Alanine Aminotransferase (U/L)"
-    settings[["measure_values"]][["AST"]]<-"Aspartate Aminotransferase (U/L)"
-    settings[["measure_values"]][["TB"]]<-"Bilirubin (umol/L)"
-    settings[["measure_values"]][["ALP"]]<-"Alkaline Phosphatase (U/L)"
+    potential_settings[["id_col"]]<-"USUBJID"
+    potential_settings[["value_col"]]<-"AVAL"
+    potential_settings[["measure_col"]]<-"PARAM"
+    potential_settings[["normal_col_low"]]<-"A1LO"
+    potential_settings[["normal_col_high"]]<-"A1HI"
+    potential_settings[["studyday_col"]]<-"ADY"
+    potential_settings[["visit_col"]]<-"VISIT"
+    potential_settings[["visitn_col"]]<-"VISITNUM"
+    potential_settings[["measure_values"]][["ALT"]]<-"Alanine Aminotransferase (U/L)"
+    potential_settings[["measure_values"]][["AST"]]<-"Aspartate Aminotransferase (U/L)"
+    potential_settings[["measure_values"]][["TB"]]<-"Bilirubin (umol/L)"
+    potential_settings[["measure_values"]][["ALP"]]<-"Alkaline Phosphatase (U/L)"
   }
   
   if(tolower(standard)=="sdtm"){
-    settings[["id_col"]]<-"USUBJID"
-    settings[["value_col"]]<-"STRESN"
-    settings[["measure_col"]]<-"TEST"
-    settings[["normal_col_low"]]<-"STNRLO"
-    settings[["normal_col_high"]]<-"STNRHI"
-    settings[["studyday_col"]]<-"DY"
-    settings[["visit_col"]]<-"VISIT"
-    settings[["visitn_col"]]<-"VISITNUM"
-    settings[["measure_values"]][["ALT"]]<-"Aminotransferase, alanine (ALT)"
-    settings[["measure_values"]][["AST"]]<-"Aminotransferase, aspartate (AST)"
-    settings[["measure_values"]][["TB"]]<-"Total Bilirubin"
-    settings[["measure_values"]][["ALP"]]<-"Alkaline phosphatase (ALP)"
+    potential_settings[["id_col"]]<-"USUBJID"
+    potential_settings[["value_col"]]<-"STRESN"
+    potential_settings[["measure_col"]]<-"TEST"
+    potential_settings[["normal_col_low"]]<-"STNRLO"
+    potential_settings[["normal_col_high"]]<-"STNRHI"
+    potential_settings[["studyday_col"]]<-"DY"
+    potential_settings[["visit_col"]]<-"VISIT"
+    potential_settings[["visitn_col"]]<-"VISITNUM"
+    potential_settings[["measure_values"]][["ALT"]]<-"Aminotransferase, alanine (ALT)"
+    potential_settings[["measure_values"]][["AST"]]<-"Aminotransferase, aspartate (AST)"
+    potential_settings[["measure_values"]][["TB"]]<-"Total Bilirubin"
+    potential_settings[["measure_values"]][["ALP"]]<-"Alkaline phosphatase (ALP)"
+  }
+  
+  
+  if(partial) {
+    
+    settings_names <- names(settings)
+    
+    potential_names <- names(potential_settings)
+    
```

Comment:
I think this looks good for now, but we should really refactor the data mapping process. Basically just need to create a data frame to track the standard, setting key and setting value and then pull values from that data set (instead of hard coding them here). That approach should let us simplify this code quite a bit.

Reply:
Created #115 to cover this at some point.

Reply:
good call, that makes perfect sense.